src/components/modals/ScreenshotModal/ScreenshotModal.component.jsx

```diff
@@ -8,17 +8,73 @@ import { Copy32Icon, Save24Icon } from '../../../icons';
 import { loginHandler } from '../SignInModal';
 import { useAuthContext } from '../../../contexts';
 import Toolbar from '../../scenegraph/Toolbar';
+import { db, storage } from '../../../services/firebase';
+import html2canvas from 'html2canvas';
+import { getDownloadURL, ref, uploadBytes } from 'firebase/storage';
+import {
+  collection,
+  doc,
+  getDoc,
+  serverTimestamp,
+  updateDoc
+} from 'firebase/firestore';
 
 function ScreenshotModal({ isOpen, onClose }) {
   const storedScreenshot = localStorage.getItem('screenshot');
   const parsedScreenshot = JSON.parse(storedScreenshot);
   const { currentUser } = useAuthContext();
-  const saveScreenshot = (value) => {
+  const saveScreenshot = async (value) => {
     const screenshotEl = document.getElementById('screenshot');
     screenshotEl.play();
     screenshotEl.setAttribute('screentock', 'type', value);
     screenshotEl.setAttribute('screentock', 'takeScreenshot', true);
+    try {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      const aframeScene = document.getElementById('screentock-destination');
+      const canvas = await html2canvas(aframeScene);
+      const resizedCanvas = document.createElement('canvas');
+      resizedCanvas.width = 320;
+      resizedCanvas.height = 240;
+      const context = resizedCanvas.getContext('2d');
+
+      context.drawImage(canvas, 0, 0, 320, 240);
```

Comment:
Instead of converting img element to canvas via html2canvas and then drawing in another canvas; in theory we can pass the screentock-destination img directly. (`drawImage()` should accept `HTMLImageElement`)

So this could be:
`     context.drawImage(aframeScene, 0, 0, 320, 240);`